virttest/utils_misc.py

```diff
@@ -851,6 +851,43 @@ def archive_as_tarball(source_dir, dest_dir, tarball_name=None,
     tarball.close()
 
 
+def get_guest_service_status(session, service, service_former=None):
+    """
+    Get service's status in guest.
+    Return 'active' for 'running' and 'active' case.
+    Return 'inactive' for 'stopped' and 'inactive' case.
+
+    :param session: An Expect or ShellSession instance to operate on
+    :type devices: Expect or ShellSession class
+    :param service: Service name that we want to check status.
+    :type service: String
+    :param service_former: Service's former name
+    :type service: String
+    :return: 'active' or 'inactive'
+    :rtype: String
+    """
+    try:
+        session.cmd("systemctl --version")
+        cmd = "systemctl status %s.service" % service
+        output = session.cmd_output(cmd)
+    except Exception:
+        if service_former is not None:
+            cmd = "service %s status" % service_former
+            output = session.cmd_output(cmd)
+        else:
+            raise exceptions.TestError("Fail to get '%s' service status.\n%s" %
+                                       (service, output))
```

Comment:
Define cmd 'output' before the try.except block, then you can avoid arise TestError here